src/platform/w3dengine/client/w3dbridgebuffer.cpp

```diff
@@ -0,0 +1,790 @@
+    Ref_Ptr_Release(m_bridgeTexture);
+    Ref_Ptr_Release(m_leftMesh);
+    Ref_Ptr_Release(m_sectionMesh);
+    Ref_Ptr_Release(m_rightMesh);
+    char texture[260];
+    char model[260];
+    char left[260];
+    char section[260];
+    char right[260];
+    memset(texture, 0, sizeof(texture));
+    memset(model, 0, sizeof(model));
+    strcpy(texture, "No Texture");
+    strcpy(model, "BRIDGESECTIONAL");
+    float scale = 0.7f;
+    float length = 170.0f;
+    TerrainRoadType *bridge = g_theTerrainRoads->Find_Bridge(m_template);
+
+    if (!bridge) {
+        return false;
+    }
+
+    scale = bridge->Get_Bridge_Scale();
+
+    switch (cur_damage_state) {
+        case BODY_PRISTINE:
+            strcpy(texture, bridge->Get_Texture());
+            strcpy(model, bridge->Get_Bridge_Model());
+            break;
+        case BODY_DAMAGED:
+            strcpy(texture, bridge->Get_Texture_Damaged());
+            strcpy(model, bridge->Get_Bridge_Model_Damaged());
+            break;
+        case BODY_REALLYDAMAGED:
+            strcpy(texture, bridge->Get_Texture_Really_Damaged());
+            strcpy(model, bridge->Get_Bridge_Model_Really_Damaged());
+            break;
+        case BODY_RUBBLE:
+            strcpy(texture, bridge->Get_Texture_Broken());
+            strcpy(model, bridge->Get_Bridge_Model_Broken());
+            break;
+    }
+
+    strcpy(left, model);
+    strcat(left, ".BRIDGE_LEFT");
+    strcpy(section, model);
+    strcat(section, ".BRIDGE_SPAN");
+    strcpy(right, model);
+    strcat(right, ".BRIDGE_RIGHT");
+    m_bridgeTexture = W3DAssetManager::Get_Instance()->Get_Texture(texture, MIP_LEVELS_3);
+    m_leftMtx.Make_Identity();
+    m_rightMtx.Make_Identity();
+    m_sectionMtx.Make_Identity();
+    RenderObjClass *obj = W3DAssetManager::Get_Instance()->Create_Render_Obj(model);
+
+    if (obj) {
+        for (int i = 0; i < obj->Get_Num_Sub_Objects(); i++) {
+            RenderObjClass *subobj = obj->Get_Sub_Object(i);
+            Matrix3D m = subobj->Get_Transform();
+
+            if (!strncasecmp(left, subobj->Get_Name(), strlen(left))) {
+                m_leftMtx = m;
+                strcpy(left, subobj->Get_Name());
+            }
+
+            if (!strncasecmp(section, subobj->Get_Name(), strlen(section))) {
+                m_sectionMtx = m;
+                strcpy(section, subobj->Get_Name());
+            }
+
+            if (!strncasecmp(right, subobj->Get_Name(), strlen(right))) {
+                m_rightMtx = m;
+                strcpy(right, subobj->Get_Name());
+            }
+
+            Ref_Ptr_Release(subobj);
+        }
+
+        Ref_Ptr_Release(obj);
+        m_leftMesh = (MeshClass *)W3DAssetManager::Get_Instance()->Create_Render_Obj(left);
+        m_sectionMesh = (MeshClass *)W3DAssetManager::Get_Instance()->Create_Render_Obj(section);
+        m_rightMesh = (MeshClass *)W3DAssetManager::Get_Instance()->Create_Render_Obj(right);
+        m_scale = scale;
+
+        if (m_leftMesh) {

```

Comment:
add explicit nullptrs to all of these m_...

Reply:
fixed.